docs/django/image_field.md

```diff
@@ -0,0 +1,147 @@
+# Django Image field
+
+For the ImageField to work it's necessary to have the Pillow library installed.
+
+## Django Models
+
+The setup of the model containing the image field can be done in the following way.
+
+````python
+class BookInformation(models.Model):
+    #...
+    cover_image = models.ImageField(upload_to="covers/", null=True)
+    #...
+````
+Note that the `upload_to` option corresponds to the path in which the images are stored/saved in the media folder.
+In this case the path will be `.../library_project/media/covers/`.
+
+## Factory Boy mock data
+
+The easiest way to generate mock images data with factory boy is with the `factory.django.ImageField()` method just like the example below.
+
+```` python 
+class BookInformationFactory(factory.django.DjangoModelFactory):
+    #...
+    cover_image = factory.django.ImageField(color=factory.Faker("color"))
+    #...
+````
+
+Note that the `color` option can Hardcoded with something like `color="blue"` and that will result in the 
+generation of images with the color blue. In this case, `factory.Faker("color")` will generate random color names
+which will result in the generation of images with diferent colors for each `cover_image`.
+
+## Django Model Serializer 
+
+Considering the previously shown factory, here is the correspondent serializer for POST and PUT requests.  
+
+````python 
+class BookInformationCreateUpdateSerializer(serializers.ModelSerializer):
+    cover_image = serializers.ImageField(required=False, allow_empty_file=True)
+
+    class Meta:
+        model = BookInformation
+        fields = [     
+        #... , 
+        "cover_image", 
+        #... ,
+        ]
+````
+
+## API View
+
+````python
+class BookInformationViewset(viewsets.ModelViewSet):
+    queryset = BookInformation.objects.all()
+    serializer_class = BookInformationCreateUpdateSerializer
+````
+
+## ImageField Views Tests
+
+To test the views that involve the `cover_image` attribute you can simply generate an image based on the
+implemented factory for the cover image just like the example below.
+
+````python
+from PIL import Image
+import shutil
+from ..fixtures import REMOVE_MEDIA_FILES_PATH
+
+class BookCreateViewTestCase(APITestCase):
+    def setUp(self):
+        #...
+        self.data = {
+            "cover_image": #...
+            #...
+        }
+
+    def tearDown(self):
+        shutil.rmtree(REMOVE_MEDIA_FILES_PATH)
+
+    def test_create_with_image(self):
+        cover = BookInformationFactory.create().cover_image
```

Comment:
why do you need the `.create()`? what's the difference from just `BookInformationFactory()`?

Reply:
The ``create()`` method will create an instance in the DB and in this case we will store the image generated to be later compared.

Reply:
Functionality wise it's exactly the same, the only reason to do `.create()` is to clear any ambiguity about the `Factory()` constructor strategy:

The [Factory](https://factoryboy.readthedocs.io/en/stable/reference.html#factory.Factory) class provides a few methods for getting objects; the usual way being to simply call the class:
```python
UserFactory()               # Calls UserFactory.create()
UserFactory(login='john')   # Calls UserFactory.create(login='john')
```
Under the hood, factory_boy will define the [Factory](https://factoryboy.readthedocs.io/en/stable/reference.html#factory.Factory) [__new__()](https://docs.python.org/3/reference/datamodel.html#object.__new__) method to call the default [strategy](https://factoryboy.readthedocs.io/en/stable/reference.html#strategies) of the [Factory](https://factoryboy.readthedocs.io/en/stable/reference.html#factory.Factory).
[source](https://factoryboy.readthedocs.io/en/stable/reference.html#factory.Factory)

(...) factory_boy supports two main strategies for generating instances, plus.
The ‘create’ strategy builds and saves an instance into its appropriate datastore. **This is the default strategy of factory_boy;**
[source](https://factoryboy.readthedocs.io/en/stable/reference.html#strategies)